homeassistant/components/deconz/climate.py

```diff
@@ -68,7 +74,9 @@ def hvac_mode(self):
 
         Need to be one of HVAC_MODE_*.
         """
-        if self._device.on:
+        if self._device.mode in SUPPORT_HVAC:
+            return self._device.mode
+        elif self._device.state_on:
```

Comment:
You don't need the elif here since you return if the previous statement evaluates to true